akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/BasicPersistentBehaviorCompileOnly.scala

```diff
@@ -76,91 +78,106 @@ object BasicPersistentBehaviorCompileOnly {
-      context.log.info2("Snapshot actor {} => state: {}", context.self.path.name, state)
-      true
-    })
+  object RecoveryBehavior {
+    //#recovery
+    def apply(): Behavior[Command] =
+      EventSourcedBehavior[Command, Event, State](
+        persistenceId = PersistenceId("abc"),
+        emptyState = State(),
+        commandHandler = (state, cmd) => throw new RuntimeException("TODO: process the command & return an Effect"),
+        eventHandler = (state, evt) => throw new RuntimeException("TODO: process the event return the next state"))
+        .receiveSignal {
+          case (state, RecoveryCompleted) =>
+            throw new RuntimeException("TODO: add some end-of-recovery side-effect here")
+        }
+    //#recovery
   }
-  //#wrapPersistentBehavior
 
-  //#supervision
-  val supervisedBehavior = samplePersistentBehavior.onPersistFailure(
-    SupervisorStrategy.restartWithBackoff(minBackoff = 10.seconds, maxBackoff = 60.seconds, randomFactor = 0.1))
-  //#supervision
+  object TaggingBehavior {
+    //#tagging
+    def apply(): Behavior[Command] =
+      EventSourcedBehavior[Command, Event, State](
+        persistenceId = PersistenceId("abc"),
+        emptyState = State(),
+        commandHandler = (state, cmd) => throw new RuntimeException("TODO: process the command & return an Effect"),
+        eventHandler = (state, evt) => throw new RuntimeException("TODO: process the event return the next state"))
+        .withTagger(_ => Set("tag1", "tag2"))
+    //#tagging
+  }
 
-  // #actor-context
-  import akka.persistence.typed.scaladsl.Effect
-  import akka.persistence.typed.scaladsl.EventSourcedBehavior.CommandHandler
-
-  val behaviorWithContext: Behavior[String] =
-    Behaviors.setup { context =>
-      EventSourcedBehavior[String, String, State](
-        persistenceId = PersistenceId("myPersistenceId"),
-        emptyState = new State,
-        commandHandler = CommandHandler.command { cmd =>
-          context.log.info("Got command {}", cmd)
-          Effect.persist(cmd).thenRun { state =>
-            context.log.info("event persisted, new state {}", state)
-          }
-        },
-        eventHandler = {
-          case (state, _) => state
+  object WrapBehavior {
+    //#wrapPersistentBehavior
+    def apply(): Behavior[Command] = Behaviors.setup { context =>
+      EventSourcedBehavior[Command, Event, State](
+        persistenceId = PersistenceId("abc"),
+        emptyState = State(),
+        commandHandler = (state, cmd) => throw new RuntimeException("TODO: process the command & return an Effect"),
+        eventHandler = (state, evt) => throw new RuntimeException("TODO: process the event return the next state"))
+        .snapshotWhen((state, _, _) => {
+          context.log.info2("Snapshot actor {} => state: {}", context.self.path.name, state)
+          true
         })
     }
-  // #actor-context
+    //#wrapPersistentBehavior
+  }
+
+  object Supervision {
+    //#supervision
```

Comment:
Should this be below the `apply()`?

Reply:
I included apply on purpose, but might be better to not, so I moved it in all places here